pool/src/pool.rs

```diff
@@ -218,12 +239,18 @@ impl Pool {
 			txs.push(entry.tx.clone());
 
 			let tx = transaction::aggregate(txs)?;
-			tx.validate(self.verifier_cache.clone())?;
+
+			// TODO - Is this necessary? We validate_raw_tx below.
+			// Validate this single aggregated tx (existing pool + new tx),
+			// not subject to tx weight limits.
+			// tx.validate(WeightVerificationType::AsPool, self.verifier_cache.clone())?;
+
 			tx
 		};
 
-		// Validate aggregated tx against a known chain state.
-		self.validate_raw_tx(&agg_tx, header)?;
+		// Validate aggregated tx (existing pool + new tx), ignoring tx weight limits.
+		// Validate against known chain state at the provided header.
+		self.validate_raw_tx(&agg_tx, header, WeightVerificationType::AsPool)?;
```

Comment:
This is the critical change here for the PR. 
When we add a new tx to the pool we want to aggregate the entire contents of the pool and check the resulting tx is still valid.
But we do _not_ want to verify that the weight here is below `MAX_BLOCK_WEIGHT`.